src/map/lua/lua_item.cpp

```diff
@@ -370,6 +370,31 @@ void CLuaItem::setExData(sol::table const& newData)
     }
 }
 
+auto CLuaItem::getFishData() -> sol::table
+{
+    if (!fishingutils::IsFish(m_PLuaItem))
+    {
+        ShowError("ERROR: Attempted to retrieve fish data from non-fish item.");
+        return sol::lua_nil;
+    }
+
+    if (auto PFish = static_cast<CItemFish*>(m_PLuaItem))
```

Comment:
Using static_cast here means that ANY non nullptr pointer will pass this check and garbage will go inside the if statement

Reply:
This may disappear when I push the updates, but I've removed this function completely since I've realized that I can't cast to a fish item.  Instead, I'm just using the getExData function and working with the data in lua directly instead.

Reply:
> I can't cast to a fish item. 
Would this work?

https://github.com/LandSandBoat/server/pull/5704/files

Reply:
Winter:  I just saw that now.  It probably _would_ work, but I've removed any casting to a `CFishItem` right out of the code and just use the ExData functions.  When I wrote this on Horizon, we didn't have the `getExData` and `setExData`, but those work perfectly for this.